src/components/AccountComponents/UserInfo.js

```diff
@@ -1,43 +1,58 @@
-import React, { Component, useState } from 'react'
-import UserService from '../../services/UserService'
-import Button from 'react-bootstrap/Button'
-import Collapse from 'react-bootstrap/Collapse'
+import React, { Component, useState } from 'react';
+import UserService from '../../services/UserService';
+import Button from 'react-bootstrap/Button';
+import Collapse from 'react-bootstrap/Collapse';
+import jwt_decode from 'jwt-decode';
+import { withRouter } from "react-router";
+import { Redirect } from 'react-router-dom';
 
-export default class UserInfo extends Component {
+class UserInfo extends Component {
   constructor(props) {
     super(props);
     this.state = {
         user: {},
         user_loaded: false,
+        redirect: false,
     }
   }
+  //loads in user
+  //keeps user fields hidden when the user cannot be loaded
   componentDidMount() {
     try{
-      UserService.read().then(res => {
+      UserService.get().then(res => {
         this.setState({user: res.data});
         this.setState({user_loaded: true});
       }).catch((error)=>{
         console.log(error);
+        this.setState({user_loaded: false});
       });
     }
     catch(e){
       console.log(e);
     }
     
   }
-  closeAccount() {
-    try{
-      UserService.delete().then(resp => {
-        this.props.history.push('/login');
-      }).catch((e)=>{
-        console.log(e);
-      });
-    }
-    catch(e){
-      console.log(e);
-    }
-  }
   render (){
+    //exits to root page on successful closing of the account
+    function closeAccount() {
```

Comment:
Styling: Why are you declaring functions in the render function and not outside of it?